File: app/src/main/java/com/sjn/taggingplayer/db/dao/PlaylistDao.java
package com.sjn.taggingplayer.db.dao;

import com.sjn.taggingplayer.db.Playlist;
import com.sjn.taggingplayer.db.PlaylistSong;

import io.realm.Realm;

public class PlaylistDao extends BaseDao {

    private static PlaylistDao sInstance;

    public static PlaylistDao getInstance() {
        if (sInstance == null) {
            sInstance = new PlaylistDao();
        }
        return sInstance;
    }

    public Playlist save(Realm realm, Playlist rawPlaylist) {
        realm.beginTransaction();
        rawPlaylist.setId(getAutoIncrementId(realm, Playlist.class));
        int id = getAutoIncrementId(realm, PlaylistSong.class);
        for (PlaylistSong playlistSong : rawPlaylist.getSongs()) {
            playlistSong.setId(id++);
        }
        realm.commitTransaction();
        return rawPlaylist;
    }
}

File: app/src/main/java/com/sjn/stamp/db/dao/PlaylistDao.kt
package com.sjn.stamp.db.dao

import com.sjn.stamp.db.Playlist
import com.sjn.stamp.db.PlaylistSong

import io.realm.Realm

@Suppress("unused")
object PlaylistDao : BaseDao() {

    fun save(realm: Realm, rawPlaylist: Playlist): Playlist {
        realm.beginTransaction()
        rawPlaylist.id = getAutoIncrementId(realm, Playlist::class.java)
        var id = getAutoIncrementId(realm, PlaylistSong::class.java)!!.toInt()
        for (playlistSong in rawPlaylist.songs!!) {
            playlistSong.id = id++.toLong()
        }
        realm.commitTransaction()
        return rawPlaylist
    }
}